youtube_dl/extractor/srgssr.py

```diff
@@ -33,102 +50,167 @@ def _get_tokenized_src(self, url, video_id, format_id):
             video_id, 'Downloading %s token' % format_id, fatal=False) or {}
         auth_params = token.get('token', {}).get('authparams')
         if auth_params:
-            url += '?' + auth_params
+            url += ('?' if '?' not in url else '&') + auth_params
         return url
 
-    def get_media_data(self, bu, media_type, media_id):
-        media_data = self._download_json(
-            'http://il.srgssr.ch/integrationlayer/1.0/ue/%s/%s/play/%s.json' % (bu, media_type, media_id),
-            media_id)[media_type.capitalize()]
-
-        if media_data.get('block') and media_data['block'] in self._ERRORS:
-            message = self._ERRORS[media_data['block']]
-            if media_data['block'] == 'GEOBLOCK':
+    def _get_media_data(self, bu, media_type, media_id):
+        query = {'onlyChapters': True} if media_type == 'video' else {}
+        full_media_data = self._download_json(
+            'https://il.srgssr.ch/integrationlayer/2.0/%s/mediaComposition/%s/%s.json'
+            % (bu, media_type, media_id),
+            media_id, query=query)['chapterList']
+        try:
+            media_data = next(
+                x for x in full_media_data if x['id'] == media_id)
+        except StopIteration:
+            raise ExtractorError('No media information found')
+
+        if media_data.get('blockReason') and media_data['blockReason'] in self._ERRORS:
+            message = self._ERRORS[media_data['blockReason']]
+            if media_data['blockReason'] == 'GEOBLOCK':
                 self.raise_geo_restricted(
                     msg=message, countries=self._GEO_COUNTRIES)
             raise ExtractorError(
                 '%s said: %s' % (self.IE_NAME, message), expected=True)
 
         return media_data
 
-    def _real_extract(self, url):
-        bu, media_type, media_id = re.match(self._VALID_URL, url).groups()
+    def _get_subtitles(self, media_data, bu, media_type):
+        subtitles = {}
+        if media_type == 'audio':
+            return subtitles
+
+        subtitle_data = media_data.get('subtitleList', [])
+        default_language_codes = {
+            'srf': 'de',
+            'rts': 'fr',
+            'rsi': 'it',
+            'rtr': 'rm',
+            'swi': 'en',
+        }
+        known_formats = ('TTML', 'VTT')
```

Comment:
do you have an example with TTML subtitles? all the videos that i've tested with has only VTT subtitles.

Reply:
I think TTML subtitles do not exist anymore.

This video (https://www.srf.ch/play/tv/rundschau/video/schwander-rot-gruene-stadtpolitik-min-li-marti-tamilen-kirche?urn=urn:srf:video:2da578e3-dbb4-4657-a539-f01089a67831) used to have TTML subtitles.